### ISSUE TITLE
Update resets columns

### ISSUE THREAD

**Issue by mneuburg-livefront** [losnoco/Cog#376]
Whenever there's an update to Cog, downloading and installing the update from within Cog resets my column settings to the default.

Example: Before the update:

![Screen Shot 2023-09-03 at 8 53 52 AM](https://github.com/losnoco/Cog/assets/125824749/c8e6c985-d129-4b25-af68-6f896c555844)

After the update (just now):

![Screen Shot 2023-09-03 at 8 54 27 AM](https://github.com/losnoco/Cog/assets/125824749/deaa0983-87eb-4767-824f-c3569537d08d)

Please make Cog stop doing that. Thank you!

**Comment by kode54** [comment]
This seems to be a bug with the UserDefaults container that is used to store these settings. I have no idea why it's doing it, either. It doesn't seem to do it on any more recent macOS versions, either. Anything Big Sur or newer doesn't seem to be affected.

Apple is also making it harder to debug this, by removing Rosetta debugging from Xcode.

**Comment by mneuburg-livefront** [comment]
Agreed, I should have mentioned that this is High Sierra. Let me know if I can do more to help debug, such as looking in the prefs file.

**Comment by kode54** [comment]
I don't know that it's the prefs file at all. The issue seems to happen on anything older than Big Sur (11.0), and I don't know how long it's been going on.

**Comment by mneuburg-livefront** [comment]
It seems that this might have something to do with restarting the computer. I was using Cog just fine, but then because of a schedule power outage I shut down the computer. When the power outage was over and I started up the computer again, Cog had reset its columns to the defaults. So it seems as if Cog is not writing out the column settings to persistent storage in a way that it can use to reconstruct them when relaunched from a cold start. Sorry, I know I'm just hand-waving, but I'm trying to give as much info as I can, when I encounter it.

**Comment by kode54** [comment]
This was a bug with column state memory, which I have rewritten to not use system methods of saving the playlist state.
